### PR TITLE
Make sure dep is a no-op on every PR

### LINKED ISSUE #77: Add a step to verify vendor code before running unit tests.
## Expected Behavior
On a Branch, we would want developers to run dep ensure, so that unused dependencies are removed.
We can do this by running a [script](https://github.com/grafeas/kritis/blob/master/hack/dep.sh) in the checks which runs dep ensure and check if they are any changes to vendor dir.

## Actual Behavior
Right now, developers can easily add unused deps and overlook the libs pulled in since they tend to do this
```
git add vendor/
```

## Steps to Reproduce the Problem
none

### PR DESCRIPTION
**Pull request by bobcatfish**
Literally copy pasted the logic from the build presubmit script:
https://github.com/knative/build/blob/5f0b563ce9064e49530b2e209b17b4d1bd71e99b/test/presubmit-tests.sh#L23
to run `verify-codegen.sh` to make sure that `dep` will be a no-op.
We've been seeing different behaviour from `update-codegen.sh` on
different ppl's machines, but at least this way we can be consistent
with what we see on Prow.

Also copying the code to make sure building is successful (especially
valuable before we have integration test automation; at least stop them
from completely breaking) and verify our licenses.

Fixes #77

### CODE REVIEW

**Comment by bobcatfish** [comment]
Need to wait and see what happens with `pull-knative-build-pipeline-build-tests `

**Comment by bobcatfish** [comment]
image

i guess this is why we aren't running with `set -x`

i'm going to make a PR to update those header functions in test-infra not to use `set -x`, i suggest we go ahead with this anyway

image

Seems like it's working i guess?

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -16,83 +16,34 @@
...
+
 function build_tests() {
-  echo "Running build tests"
-  # TODO(aaron-prindle) add build tests
-  echo "TODO(aaron-prindle) add build tests"
+  header "Verifying code structure and contents"
+
+  local result=0
+  subheader "Checking that we can build the code successfully"
+  go build -v ./... || result=1
```
Comment (tejal29) [inline test/presubmit-tests.sh]: Do you also want to #77 in this PR?

**Comment by tejal29** [comment]
Sorry /approve

File: test/presubmit-tests.sh (hunk 1 above)
Comment (bobcatfish) [reply test/presubmit-tests.sh]: sure lemme see - as long as `go test` won't return an error when there are no tests yet

**Comment by bobcatfish** [comment]
image

@tejal29 i think what i need is `/lgtm`, since im an OWNER i get approval automatically. note tho that that would make `tide` pick this up and merge it right away so ill add a `/hold` to prevent that.

I'm gonna add #77 too tho and then ill need another `/lgtm`

/hold

**Comment by bobcatfish** [comment]
/hold cancel

Okay runs the unit tests now too! Sad side note, running the presubmit script will now require `bazel` b/c it's being used to parse test results @_@ I'll add that to the docs i'm adding in the integration test PR.

**Comment by bobcatfish** [comment]
oh boy, flakey already

```
W0928 20:37:25.454] 	(1) failed to list versions for https://go.googlesource.com/crypto: fatal: remote error: Access denied to IP 35.224.139.156
```

**Comment by adrcunha** [comment]
I don't know why I'm always CCed in PRs for this repo, but...

I happened to catch a few questions from @bobcatfish, so answering them here:

> running the presubmit script will now require bazel b/c it's being used to parse test results @_@

As Nader pointed out, bazel is only used in Prow jobs so Gubernator and Testgrid show individual test results. Locally, just `go test` is used.

> i'm going to make a PR to update those header functions in test-infra not to use set -x

The header functions are not doing anything with set. You're getting a verbose output because of

https://github.com/knative/build-pipeline/blob/master/test/presubmit-tests.sh#L19

Sounds either like debugging, or because your team likes verbose output. If you have a preference for verbose output, make the changes in your scripts here. The scripts in `test-infra` by design don't have verbose output because that causes too much unnecessary information in test logs, specially for E2E tests.

**Comment by adrcunha** [comment]
> You're getting a verbose output because of
> https://github.com/knative/build-pipeline/blob/master/test/presubmit-tests.sh#L19

I didn't notice that you were the one adding that. :P

**Pull request merged**